### PR TITLE
api management events schema

### PR DESCRIPTION
**Pull request by KacieKK**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      "type": "object",
+      "properties": {
+        "resourceUri": {
+          "description": "The fully qualified ID of the resource that the compliance state change is for, including the resource name and resource type. Uses the format, `/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroup>/Microsoft.ApiManagement/service/<ServiceName>/<ResourceType>/<ResourceName>`",
+          "type": "string"
+        }
+      }
+    },
+    "APIUpdatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.APIUpdated event.",
```
Comment (ellismg) [inline specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json:111]: I noticed that for some events `API` is capitalized, and in other cases, it's spell `Api` (e.g. `APIUpdated` vs `ApiReleaseUpdated`)  I just want to confirm that the difference in casing between these event is expected.

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Azure API Management events published to Azure Event Grid",
+    "description": "Describes the schema of the Azure API Management events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "UserCreatedEventData": {
```
Comment (ellismg) [inline specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json:10]: We use the names of these `EventData` definitions for the corresponding model types in each language.  Some of these names are fairly general, and could be similar to say events from other sources (for example, I think ARM publishes similar events about subscriptions).

For the ones that don't already have an Api prefix, could we add something like `ApiManagement`, e.g. `ApiManagementUserCreatedEventData` and `ApiManagementSubscriptionCreatedEventData`

**Review by ellismg — CHANGES_REQUESTED** [review]

**Comment by KacieKK** [comment]
we're failing Avocado and prettier check because of the example, which is required to be a "real" example. Please advise on this.

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apiUpdated.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "id": "95015754-aa51-4eb6-98d9-9ee322b82ad7",
+  "topic": "/subscriptions/{subscription-id}/resourceGroups/{your-rg}/providers/Microsoft.ApiManagement/service/{your-APIM-instance}",
+  "subject": "/apis/myapi;Rev=1",
+  "data": {
+    "resourceUri": "/subscriptions/subscription-id}/resourceGroups/{your-rg}/providers/Microsoft.ApiManagement/service/{your-APIM-instance}/apis/myapi;Rev=1"
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apiUpdated.json:6]:
```suggestion
    "resourceUri": "/subscriptions/{subscription-id}/resourceGroups/{your-rg}/providers/Microsoft.ApiManagement/service/{your-APIM-instance}/apis/myapi;Rev=1"
```

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json (hunk 1 above)
Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json:111]: Also, curious about this - We would likely need to rename this via swagger transforms in .NET. It also seems odd that we have API (in whatever casing) twice in the name of the events.

**Comment by KacieKK** [comment]
Could you elaborate a bit on “have api twice?"  Not quite sure what this
means. Apimanagement is the name of our service.

**Comment by JoshLove-msft** [comment]
> Could you elaborate a bit on “have api twice?" Not quite sure what this means. Apimanagement is the name of our service.

So for ApiManagementAPIUpdatedEventData, is it intentional that we have both "Api" and "API" in the name? I'm curious what the second API is meant to indicate.

**Comment by KacieKK** [comment]
Yes we did it on purpose, apimanagement is the name of our service, api is
an entity under that. check this here
https://docs.microsoft.com/en-us/rest/api/apimanagement/2020-12-01/apis

**Comment by JoshLove-msft** [comment]
> Yes we did it on purpose, apimanagement is the name of our service, api is an entity under that. check this here https://docs.microsoft.com/en-us/rest/api/apimanagement/2020-12-01/apis

I see. Is there a reason the casing is different? I'd expect it to be "Api" in both places.

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json (hunk 1 above)
Comment (lmazuel) [reply specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json:111]: The service is ApiManagement, and the event is that an API has been updated, so it actualle seems normal to me?

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/APIManagement.json:111]: Yeah, that makes sense but the casing should be consistent, right?

**Review by lmazuel — APPROVED** [review]

**Review by ellismg — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
> Renaming "ApiManagementAPI" to "ApiManagementApi" requires an extra rollout train to all our services. We prefer to stay as it is.

Aren't there already other changes from this PR that will require another rollout?

**Comment by JoshLove-msft** [comment]
> > Renaming "ApiManagementAPI" to "ApiManagementApi" requires an extra rollout train to all our services. We prefer to stay as it is.
> 
> Aren't there already other changes from this PR that will require another rollout?

Actually, it looks like the casing has already been updated to "Api" everywhere, so I think this no longer an issue?

**Comment by JoshLove-msft** [comment]
> > Renaming "ApiManagementAPI" to "ApiManagementApi" requires an extra rollout train to all our services. We prefer to stay as it is.
> 
> Aren't there already other changes from this PR that will require another rollout?

Actually, it looks like the casing has already been updated to "Api" everywhere.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by KacieKK** [comment]
@JoshLove-msft yes, I already changed it to pascal. are we good to merge?

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft yes, I already changed it to pascal. are we good to merge?

Looks like it. @lmazuel can you merge this?

File: specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apireleaseDeleted.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "id": "95015754-aa51-4eb6-98d9-9ee322b82ad7",
+  "topic": "/subscriptions/{subscription-id}/resourceGroups/{your-rg}/providers/Microsoft.ApiManagement/service/{your-APIM-instance}",
+  "subject": "/apis/{apiId}/releases/{releaseId}",
+  "data": {
+    "resourceUri": "/subscriptions/subscription-id}/resourceGroups/{your-rg}/providers/Microsoft.ApiManagement/service/{your-APIM-instance}/apis/{apiId}/releases/{releaseId}"
+  },
+  "eventType": "Microsoft.ApiManagement.APIReleaseDeleted",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apireleaseDeleted.json:8]: Just want to check - are the event types still using "API" or were they also updated to "Api" to match the event names?

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apireleaseDeleted.json:8]: /cc @KacieKK

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ApiManagement/stable/2018-01-01/examples/apireleaseDeleted.json:8]: My mistake - I see that the event names were updated, but the event descriptions were left in the API casing. Please disregard my question.

**Pull request merged**